service/collector/app_operator.go

```diff
@@ -130,42 +146,79 @@ func (a *AppOperator) collectAppVersions(ctx context.Context) (map[string]bool,
 
 	for _, app := range l.Items {
 		version := app.Labels[label.AppOperatorVersion]
-		appVersions[version] = true
+		appNamespaces, ok := appVersions[version]
+		if !ok {
```

Comment:
```suggestion
		if ok {
		   continue
		}
```
How about this and saving indents for few lines?

Reply:
Nice that is a lot cleaner. Thx!

Reply:
This didn't work out unfortunately. As you need to define the map if it doesn't exist.

I've extracted the logic to a calculateNamespace func so less indentation is needed. Will give it a final test in the morning before merging.